### PR TITLE
Document AbstractFormFieldDecorator

### LINKED ISSUE #217: Document AbstractFormFieldDecorator
see WoltLab/WCF#4469

### PR DESCRIPTION
**Pull request by TimWolla**
Resolves #217

/cc @Hanashi

### CODE REVIEW

File: docs/php/api/form_builder/form_fields.md [hunk 1]
```diff
@@ -19,6 +19,11 @@ The methods `step($step = null)` and `getStep()` can be used to set and get the
 The default step for form fields with integer values is `1`.
 Otherwise, the default step is `any`.
 
+### `AbstractFormFieldDecorator`
+
+`AbstractFormFieldDecorator` is a default implementation of a decorator for form fields that forwards calls to all methods defined in `IFormField` to the respectie method of the decorated object.
```
Comment (TimWolla) [inline docs/php/api/form_builder/form_fields.md]:
```suggestion
`AbstractFormFieldDecorator` is a default implementation of a decorator for form fields that forwards calls to all methods defined in `IFormField` to the respective method of the decorated object.
```

File: docs/php/api/form_builder/form_fields.md [hunk 2]
```diff
@@ -19,6 +19,11 @@ The methods `step($step = null)` and `getStep()` can be used to set and get the
...
+The class implements `IFormfield`.
+If the implementation of the more specific interfaces is required then the remaining methods must be implemented in the concrete decorator derived from `AbstractFormFieldDecorator` and the type of the `$field` property must be narrowed appropriately.
```
Comment (dtdesign) [inline docs/php/api/form_builder/form_fields.md]:
```suggestion
If the implementation of a more specific interface is required then the remaining methods must be implemented in the concrete decorator derived from `AbstractFormFieldDecorator` and the type of the `$field` property must be narrowed appropriately.
```

**Review by dtdesign — CHANGES_REQUESTED** [review]
LGTM

**Comment by Hanashi** [comment]
Looks good for me, too. Maybe you could add the patch level as extra information.

**Pull request merged**